File: app/src/main/java/com/dominikgaller/alpinebooking/booking/core/domain/tourbooking/ParticipantContact.java
package com.dominikgaller.alpinebooking.booking.core.domain.tourbooking;

import com.dominikgaller.alpinebooking.booking.core.domain.tourbooking.exception.InvalidBookingRequestException;
import java.util.Objects;

/**
 * Value object representing the contact person for a booking.
 *
 * <p>Both {@code name} and {@code email} must be non-blank.
 * No deep email format validation is performed; non-blank is sufficient.
 *
 * <p>SDD: See {@code documentation/domain/aggregate-tour-booking.spec.md}, section 2a.
 */
public record ParticipantContact(String name, String email) {

    public ParticipantContact {
        Objects.requireNonNull(name, "Contact name must not be null");
        Objects.requireNonNull(email, "Contact email must not be null");
        if (name.isBlank()) {
            throw new InvalidBookingRequestException("Contact name must not be blank");
        }
        if (email.isBlank()) {
            throw new InvalidBookingRequestException("Contact email must not be blank");
        }
    }
}

File: src/main/kotlin/com/example/contractmanagement/booking/core/domain/tourbooking/ParticipantContact.kt
package com.example.contractmanagement.booking.core.domain.tourbooking

import com.example.contractmanagement.booking.core.domain.tourbooking.exception.InvalidBookingRequestException

/**
 * Who to contact about a booking.
 *
 * Constructed from a command, so invariant violations raise a domain exception
 * (`coding-style.definition.md` § 6.2, clause one).
 *
 * SDD: see `documentation/domain/aggregate-tour-booking.spec.md`.
 */
data class ParticipantContact(
    val name: String,
    val email: String,
) {
    init {
        if (name.isBlank()) {
            throw InvalidBookingRequestException("Contact name must not be blank")
        }
        if (email.isBlank()) {
            throw InvalidBookingRequestException("Contact email must not be blank")
        }
    }
}
